src/iuse_actor.cpp

```diff
@@ -3820,3 +3821,92 @@ long mutagen_iv_actor::use( player &p, item &it, bool, const tripoint & ) const
 
     return it.type->charges_to_use();
 }
+
+iuse_actor *deploy_tent_actor::clone() const
+{
+    return new deploy_tent_actor( *this );
+}
+
+void deploy_tent_actor::load( JsonObject &obj )
+{
+    assign( obj, "radius", radius );
+    assign( obj, "wall", wall );
+    assign( obj, "floor", floor );
+    assign( obj, "floor_center", floor_center );
+    assign( obj, "door_opened", door_opened );
+    assign( obj, "door_closed", door_closed );
+    assign( obj, "broken_type", broken_type );
+}
+
+long deploy_tent_actor::use( player &p, item &it, bool, const tripoint & ) const
+{
+    int diam = 2 * radius + 1;
+
+    int dirx = 0;
+    int diry = 0;
+    if( !choose_direction( string_format( _( "Put up the %s where (%dx%d clear area)?" ), it.tname(),
+                                          diam, diam ), dirx, diry ) ) {
```

Comment:
Shouldn't it be something like `pntx` and `pnty` since there is undergoing change to result of `choose_direction` to be point instead of direction?

Reply:
Yes, I'll update this branch or the other when one of them is merged. But for now `dirx` is correct: the functions is called `choose_direction` and it returns a direction (yes, this has been changed).